File: 217ContainsDuplicate.cpp
#include<iostream>
#include<vector>
using namespace std;

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] == nums[i+1]){
                return true;
            }
        }
        return false;
    }
};

int main() {
    Solution s;
    vector<int> v = {1,2,3};
    cout << s.containsDuplicate(v) << endl;

}

File: 217ContainsDuplicate.py
class Solution(object):
    def containsDuplicate(self, nums) -> bool:
        myList = set()

        for i in nums:
            if i in myList:
                return True
            myList.add(i)
        return False

if __name__ == "__main__":
    s = Solution()
    numbers = [1,2,3,1]
    print(s.containsDuplicate(numbers))